sources/packages/backend/apps/queue-consumers/src/services/application/application.service.ts

```diff
@@ -54,4 +57,56 @@ export class ApplicationService extends RecordDataModelService<Application> {
 
     return updateResult.affected;
   }
+
+  /**
+   * Finds all applications with some pending student assessment to be
+   * processed by the workflow, ignoring applications that already have
+   * some workflows in progress or were already queued for execution.
+   * All pending student assessments will be returned ordered by its creation
+   * date to allow the select of the next one to be executed (usually only one
+   * record would be expected).
+   * @returns applications with pending assessments to be executed.
+   */
+  async getApplicationsToStartAssessments(): Promise<Application[]> {
+    // Sub query to validate if an application has assessment already being
+    // processed by the workflow.
+    const inProgressStatusesExistsQuery = this.studentAssessmentRepo
+      .createQueryBuilder("studentAssessment")
+      .select("1")
```

Comment:
I believe the understanding here for the select(1) is, the applicationId will not be returned if either of the 2 inProgressStatuses are present for the particular applicationId right?

Reply:
Yes, if the application has some assessment already in progress we will wait for it to be finished to start the next one.